### PR TITLE
HWKAPM-725 Report correct property type associated with the value

### PR DESCRIPTION
**Pull request by objectiser**

### CODE REVIEW

**Comment by objectiser** [comment]
@pavolloffay Could you review please?

**Comment by pavolloffay** [comment]
I have a question, not related to this PR but: 

`Property` can by of type: `text`, `number`, `boolean` and `binary`. But the class can store string and double. I could not find any place where `getNumber` or `setNumber` are used as it always uses only string. 

Should we do something about this? Remove `number` or use Object instead of string?

File: api/src/main/java/org/hawkular/apm/api/model/PropertyType.java [hunk 1]
```diff
@@ -41,6 +41,24 @@
...
+    Binary;
+
+    /**
+     * This method returns the property type associated with
+     * the supplied value.
+     *
+     * @param obj The property value
+     * @return The property type
+     */
+    public static PropertyType of(Object obj) {
```
Comment (pavolloffay) [inline api/src/main/java/org/hawkular/apm/api/model/PropertyType.java]: Did you consider calling this in Property constructor?

**Comment by pavolloffay** [comment]
retest this please

**Comment by objectiser** [comment]
The reason for the number field is to provide a numeric value that ES can use in queries/filters. The other approach would have been to use a custom serialiser, which is used in another situation, but at the time this was the approach taken. So its more for internal ES usage.

**Comment by objectiser** [comment]
@pavolloffay Updated to derive the property type in the constructor.

**Comment by pavolloffay** [comment]
Thanks, I was just curious because `Property.getNumber()` is never called in APM code.

```
hawkular-apm git:(HWKAPM-729-Instrumentation-test-framework-add-pull-property) g grep getNumber    
api/src/main/java/org/hawkular/apm/api/model/Property.java:120:    public Double getNumber() {
```
There is only one occurence == method definition.

File: api/src/main/java/org/hawkular/apm/api/model/PropertyType.java (hunk 1 above)
Comment (pavolloffay) [inline api/src/main/java/org/hawkular/apm/api/model/PropertyType.java]: Well this can assure that the type will be set correctly no?

**Comment by objectiser** [comment]
Its used in the json serialization used by ES.

File: api/src/main/java/org/hawkular/apm/api/model/PropertyType.java (hunk 1 above)
Comment (objectiser) [inline api/src/main/java/org/hawkular/apm/api/model/PropertyType.java]: Update has already been applied :)

**Comment by objectiser** [comment]
retest this please

**Comment by objectiser** [comment]
@pavolloffay ok to merge?

**Pull request merged**